atomic_reactor/plugins/exit_verify_media_types.py

```diff
@@ -35,10 +37,11 @@ def verify_v1_image(image, registry, log, insecure=False, dockercfg_path=None):
         log.debug("verify_v1_image: [%s] %s", r.status_code, r.url)
 
     log.debug("verify_v1_image: response headers: %s", response.headers)
-    response.raise_for_status()
-
-    # if we returned ok, then everything is fine.
-    return True
+    try:
+        response.raise_for_status()
+        return True
+    except Exception:
```

Comment:
Do we need to test the kind of exception for anything here?  expect_media originally assumed that if you were verifying a v1 image, you had a v1 image to verify and any failure was a problem.  In this case, do we need to distinguish between 404 or 503 errors (or whatever) errors?  Or can we assume that if we can't verify the V1 image for whatever reason, we can just continue the build and raise the KeyError if it was expected?

Reply:
mmm we are already using registrysession which is doing retries on (408, 500, 502, 503, 504), so I think at this point if it raises exception we can consider it as it doesn't exist

also we don't have to raise here, but just return False, as now, we are first checking what is available and after that we are actually comparing it with what is expected

logic before was calling this method ONLY when it was already expecting v1, so it made sense in raising error in that case

Reply:
Okay.  I wanted to double check the logic but that seems sensible.